pkg/controller/elasticsearch/validation/validations.go

```diff
@@ -272,11 +272,18 @@ func noDowngrades(current, proposed esv1.Elasticsearch) field.ErrorList {
 
 func validUpgradePath(current, proposed esv1.Elasticsearch) field.ErrorList {
 	var errs field.ErrorList
-	currentVer, err := version.Parse(current.Spec.Version)
+	// if available use the status version which reflects the lowest version currently running in the cluster
+	currentVer, err := version.Parse(current.Status.Version)
 	if err != nil {
-		// this should not happen, since this is the already persisted version
-		errs = append(errs, field.Invalid(field.NewPath("spec").Child("version"), current.Spec.Version, parseStoredVersionErrMsg))
+		// let's swallow that error which could be caused because we do not have a version in status yet and fall back
+		// to the Spec version which is better than nothing
```

Comment:
Should we rather check if the version is empty (`""`)? If get a parsing error I would expect we log that error explicitly as a real error.

Reply:
Absolutely. That sounds cleaner, even though I suspect it will not make much difference in practice.